chore(ci): always run `single-node` tests per PR

kwannoel opened the pull request:
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Now that the single node tests pass in `main-cron`, we should make sure they always pass, since it's a core method of deployment. https://buildkite.com/risingwavelabs/main-cron/builds/2019#018e2e3f-7131-4ea2-a34b-2bb57b866736

<!--

**Please do not leave this empty!**

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Refer to a related PR or issue link (optional)

-->

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)
<!-- To manually trigger a benchmark, please check out [Notion](https://www.notion.so/risingwave-labs/Manually-trigger-nexmark-performance-dashboard-test-b784f1eae1cf48889b2645d020b6b7d3). -->
- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.


<!--
Please create a release note for your changes.

Discuss technical details in the "What's changed" section, and
focus on the impact on users in the release note.

You should also mention the environment or conditions where the impact may occur.
-->

</details>

xxchan reviewed (commented):
<img width="1189" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/37948597/3e2771c4-1e0d-401b-8764-734192acbae7">

It's quite slow and is the bottleneck now.

BTW, is it a little wasteful to run it per PR? I feel that it can only catch `single-node` mode bug, but not logic bugs. Maybe we just need some simple tests instead of full test suite. Or maybe just run it in cron.

fuyufjh commented:
Didn't expect it to be a bottleneck. Perhaps we should revert this.

I think tests in single-node or distributed are nearly 100% the same. Keeping 1 should be enough for PR test.

yezizp2012 commented:
> Didn't expect it to be a bottleneck. Perhaps we should revert this.
> 
> I think tests in single-node or distributed are nearly 100% the same. Keeping 1 should be enough for PR test.

`single-node` can cover the implementation of the sql backend, so let's run this test in `main-cron` first.

kwannoel commented:
> Didn't expect it to be a bottleneck. Perhaps we should revert this.
> 
> I think tests in single-node or distributed are nearly 100% the same. Keeping 1 should be enough for PR test.

The test path is different. It will cover local fs store and sql backend. I have added the PR if it should be reverted regardless: https://github.com/risingwavelabs/risingwave/pull/15641.

I added it because I found there's fixes in meta store that may not get ported to the new sql backend as well.

kwannoel commented:
> BTW, is it a little wasteful to run it per PR? I feel that it can only catch single-node mode bug, but not logic bugs. Maybe we just need some simple tests instead of full test suite. Or maybe just run it in cron.

It includes logic bugs of `sql-backend` and Local fs store.
